SalesforceCommerceCloud/pwa-kit
# [commerce-sdk-react] Multisite support

This PR adds multi-site support to the `commerce-sdk-react` package.

### background

We've learned recently that SLAS tokens has the scope limited to a site, which means, you can't re-use tokens across two different site. When a user switch site, we must get new tokens from SLAS.

### Implementation

The implementation is quite simple that every time an `Auth` class is initialized, it will detect if the site is different, if site is different from the last login, it will clears the storage and attempts to restart the authentication flow.

### How can I test it?

Pull the code then

```
cd ./packages/commerce-sdk-react
npm run build
cd ../test-commerce-sdk-react
npm start

// open localhost:3000
// record the value of access_token in localstorage
// you should see the site switch at the right bottom corner, click it
// verify that the access_token is different after site switch
// verify there are network requests sent to fetch a new guest token
```

<img width="934" alt="Screen Shot 2022-10-18 at 9 31 50 AM" src="https://user-images.githubusercontent.com/10948652/196489976-d8c4d90a-5c8d-45cb-a740-c24b244a2437.png">
